system/Model.php

```diff
@@ -471,8 +471,14 @@ protected function doReplace(array $data = null, bool $returnSQL = false)
 	 */
 	protected function doErrors()
 	{
+		// $error is always ['code' => string|int, 'message' => string]
 		$error = $this->db->error();
 
+		if (intval($error['code']) === 0)
```

Comment:
```suggestion
		if ((int) $error['code'] === 0)
```

Reply:
I went with `intval()` because PHP 8 changed a lot of string conversions and comparisons. Why do you recommend the cast over the function?

Reply:
My argument is because casting is faster. For php 8 changes, as I understand it php 8 changes behavior on loose checking (`==`) only. Others are unaffected. https://3v4l.org/0EG4N